test/Overflow.hs

```diff
@@ -0,0 +1,141 @@
+getVars (Model ls) = fst <$> ls
+
+instance ToExpr (Model Symbolic)
+instance ToExpr (Model Concrete)
+
+initModel :: Model r
+initModel = Model []
+
+transition :: Model r -> Command r -> Response r -> Model r
+transition m@(Model model) cmd resp = case (cmd, resp) of
+  (Create, Created var)    -> Model ((var, 0) : model)
+  (Check _, IsNegative _)  -> m
+  (BackAndForth _ _, Unit) -> m
+  _                        -> error "impossible to transition!"
+
+precondition :: Model Symbolic -> Command Symbolic -> Logic
+precondition m cmd = case cmd of
+  Create             -> Top
+  Check var          -> var `member` getVars m
+  BackAndForth _ var -> var `member` getVars m
+
+postcondition :: Model Concrete -> Command Concrete -> Response Concrete -> Logic
+postcondition _ cmd resp = case (cmd, resp) of
+  (Create, Created _)      -> Top
+  (Check _, IsNegative bl) -> bl .== False
+  (BackAndForth _ _, Unit) -> Top
+  _                        -> Bot
+
+semantics :: Command Concrete -> IO (Response Concrete)
+semantics cmd = case cmd of
+  Create        -> Created <$> (reference . Opaque <$> newMVar 0)
+  Check var     -> do
+    val <- readMVar (opaque var)
+    return $ IsNegative $ val < 0
+  BackAndForth n var -> do
+    modifyMVar_ (opaque var) $ \x -> return $ x + fromIntegral n
+    threadDelay =<< randomRIO (0, 5000)
+    modifyMVar_ (opaque var) $ \x -> return $ x - fromIntegral n
+    return Unit
+
+mock :: Model Symbolic -> Command Symbolic -> GenSym (Response Symbolic)
+mock _ cmd = case cmd of
+  Create           -> Created <$> genSym
+  Check _var       -> return $ IsNegative False
+  BackAndForth _ _ -> return $ Unit
+
+generator :: Model Symbolic -> Maybe (Gen (Command Symbolic))
+generator (Model []) = Just (return Create)
+generator m            = Just $ frequency
+                [ (1, return Create)
+                , (3, genBackAndForth m)
+                , (3, genCheck m)
+                ]
+
+genCheck :: Model Symbolic -> Gen (Command Symbolic)
+genCheck m = do
+  Check <$> elements (getVars m)
+
+genBackAndForth :: Model Symbolic -> Gen (Command Symbolic)
+genBackAndForth m = do
+  BackAndForth 50 <$> elements (getVars m)
+
+shrinker :: Model Symbolic -> Command Symbolic -> [Command Symbolic]
+shrinker _ (BackAndForth n var) = [ BackAndForth n' var | n' <- shrink n ]
+shrinker _ _             = []
+
+sm :: StateMachine Model Command IO Response
+sm = StateMachine initModel transition precondition postcondition
+           Nothing generator shrinker semantics mock
+
+prop_sequential_overflow :: Property
+prop_sequential_overflow = forAllCommands sm Nothing $ \cmds -> monadicIO $ do
+  (hist, _model, res) <- runCommands sm cmds
+  prettyCommands sm hist (res === Ok)
+
+prop_parallel_overflow :: Property
+prop_parallel_overflow = forAllParallelCommands sm $ \cmds -> monadicIO $ do
+    prettyParallelCommandsWithOpts cmds (Just $ GraphOptions "counter_example.png" Png) =<< runParallelCommands sm cmds
```

Comment:
Very long line, perhaps put the options in a where block, i.e. where `opts = Just $ GraphOptions ...`?